rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -864,15 +865,35 @@ def get_labels_list(raw_labels_list):
             if ((not isinstance(key, text_type)) or
                     (not isinstance(value, text_type))):
                 _raise_bad_labels_list()
-            validate_inputs({'key': key, 'value': value})
-            if key.startswith(CFY_LABELS_PREFIX) and key not in CFY_LABELS:
+            parsed_key, parsed_value = _parse_label(key, value)
+            if (parsed_key.startswith(CFY_LABELS_PREFIX) and
+                    key not in CFY_LABELS):
                 _raise_labels_prefix_not_allowed()
-            labels_list.append((key.lower(), value.lower()))
+            labels_list.append((parsed_key, parsed_value))
 
     test_unique_labels(labels_list)
     return labels_list
 
 
+def _parse_label(label_key, label_value):
+    quoted_label_key = urlquote(label_key, safe='')
```

Comment:
If we want to validate that it contains only an allowed character set then we should probably validate that character set, not compare urlquoted with original.